### PR TITLE
Fixed wallet connection for unsupported chains

### PR DESCRIPTION
**Pull request by bbshark99**
Closes: https://github.com/yearn/yearn-finance-v3/issues/532

- Why this bug was happening
This issue was happening if the wallet is on the unsupported chains.

- How it's resolved
If the wallet is on the unsupported chains, it will ask to change the wallet network to the mainnet as default.
Of course, If the wallet is on the supported chains, it won't ask to change the network.

### CODE REVIEW

File: src/client/containers/Layout/index.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { Navigation, Navbar, Footer } from '@components/app';
 import { Modals, Alerts } from '@containers';
 import { getConfig } from '@config';
 import { Network, Route } from '@types';
-import { get } from '@utils';
+// import { get } from '@utils';
```
Comment (karelianpie) [inline src/client/containers/Layout/index.tsx:23]: If this is not used, let's remove it please

File: src/client/containers/Modals/CommunityThemesModal/index.tsx [hunk 2]
```diff
@@ -47,7 +47,10 @@ export interface CommunityThemesModalProps {
 
 export const CommunityThemesModal: FC<CommunityThemesModalProps> = ({ onClose, ...props }) => {
   const dispatch = useAppDispatch();
-  const { AVAILABLE_CUSTOM_THEMES, SUPPORTED_LANGS } = getConfig();
+  const {
+    AVAILABLE_CUSTOM_THEMES,
+    // SUPPORTED_LANGS,
```
Comment (karelianpie) [inline src/client/containers/Modals/CommunityThemesModal/index.tsx:52]: Remove please

**Review by karelianpie — APPROVED** [review]
LGTM, but was this tested and does it work? Looks like we are getting an instance of the SDK only for supported networks

```typescript
const sdkInstance = yearnSdk.getInstanceOf(network);
sdkInstance.context.setProvider({
  read: web3Provider.getInstanceOf(providerType),
  write: web3Provider.getInstanceOf('wallet'),
});
```

nit: Also, it would be nice to not have too many nested if/else statements

**Comment by bbshark99** [comment]
> LGTM, but was this tested and does it work? Looks like we are getting an instance of the SDK only for supported networks:
> 
> ```ts
> const sdkInstance = yearnSdk.getInstanceOf(network);
> sdkInstance.context.setProvider({
>   read: web3Provider.getInstanceOf(providerType),
>   write: web3Provider.getInstanceOf('wallet'),
> });
> ```
> 
> nit: It would be nice to not have too many nested if/else statements

correct. we are getting data from the yearnSDK if the wallet is on the supported chains

**Comment by karelianpie** [comment]
> > LGTM, but was this tested and does it work? Looks like we are getting an instance of the SDK only for supported networks:
> 
> > 
> 
> > ```ts
> 
> > const sdkInstance = yearnSdk.getInstanceOf(network);
> 
> > sdkInstance.context.setProvider({
> 
> >   read: web3Provider.getInstanceOf(providerType),
> 
> >   write: web3Provider.getInstanceOf('wallet'),
> 
> > });
> 
> > ```
> 
> > 
> 
> > nit: It would be nice to not have too many nested if/else statements
> 
> 
> 
> correct. we are getting data from the yearnSDK if the wallet is on the supported chains

My point was that we are not doing that for the unsupported wallets, so how will the SDK be instantiated?

**Comment by turtlemoji** [comment]
@bbshark99 Does the affected wallets by the network change not propagating work with fhis? That is the main problem with the wallets now

**Comment by bbshark99** [comment]
> @bbshark99 Does the affected wallets by the network change not propagating work with fhis? That is the main problem with the wallets now, not being able to change networks on all wallets because onboard doesn't trigger the network change event Ty for your contribution!

correct. If the user changes the wallet network to the unsupported network, it won't trigger the wallet network change event. do you want me to fix this?

**Comment by bbshark99** [comment]
> > @bbshark99 Does the affected wallets by the network change not propagating work with fhis? That is the main problem with the wallets now, not being able to change networks on all wallets because onboard doesn't trigger the network change event Ty for your contribution!
> 
> correct. If the user changes the wallet network to the unsupported network, it won't trigger the wallet network change event. do you want me to fix this?

@turtlemoji We are getting data from the yearnSDK but if the wallet is on unsupported chains, we are not able to get data from the yearnSDK.
image

**Comment by xgambitox** [comment]
Hey @bbshark99! Tested connecting with Wallet Connect + Rainbow, but:
1) When connected in eth mainnet, it still shows as if connected to incorrect network
image
2) Cant change to another network, either by trying through Rainbow wallet setting, or by directly trying to change through our app. It stays in eth mainnet.

**Comment by xgambitox** [comment]
@FoxTheSin ser, would be of great help to test this one on your side and let us know results, so we iterate on the solution.

**Comment by FoxTheSin** [comment]
tested with rainbow wallet on mobile.
change network on yearn.finance > opens rainbow wallet, allow, but when return to yearn.finance it doesn't change the network. still on eth

**Comment by FoxTheSin** [comment]
Rainbow doesn't support fantom, but even with Arbitrum it doesn't update yearn.finance, only stays on eth

**Comment by FoxTheSin** [comment]
Also unable to "disconnect" wallet on mobile (not sure if related)

**Comment by FoxTheSin** [comment]
relevant blocknative
https://github.com/blocknative/web3-onboard/issues/808

**Comment by xgambitox** [comment]
Closing this one in favor of #559 and/or #560

**Pull request closed without merging**